src/controllers/user.controller.js

```diff
@@ -78,4 +81,89 @@ const loginUser = async (req, res, next) => {
     }
   })(req, res, next);
 };
-export { registerUser, loginUser };
+// export { registerUser, loginUser };
+
+
+const apiKey = process.env.API_KEY;
+sgMail.setApiKey(apiKey);
+
+ const resetEmail = async (req, res) => {
+  try {
+  
+   const userEml = req.body.email;
+   const userEmail = await findUser(userEml);
+    if (userEmail == false) {
+      res.status(404).json("User not found");
+    } else {
+      const userToken = generateToken(userEmail.email, { expiresIn: '10m' });
+
+      const message = {
+        to: req.body.email,
+        from: {
+          name: "DESTRUCTORS",
+          email: process.env.SEND_EMAIL
+        },
+        subject: 'Reset password',
+
+        html: `<p stlye="font-size: 30px"><strong> Hi <br> <br>
+            Please click on this link below to reset your password:<br> ${userToken}><br>Remember, beware of scams and keep this one-time verification link confidential.<br>
+            Thanks, </strong><br> DESTRUCTORS </p>`
+
+      };
+      sgMail
+        .send(message)
+        .then((res) => console.log('email sent...'))
+        .catch((error) => {
+          console.log(error.message)
+        });
+
+      const newUser = await model.ResetPassword.create({
+        email: req.body.email,
+        token: userToken
+      });
+
+      res.status(200).json({ message: "Reset password email has been sent, check your inbox", token: userToken});
+    }
+  } catch (error) {
+    res.status(500).json({error: "Server error"});
+  }
+};
+
+ const resetPassword = async (req, res) => {
+  try {
+    const token = req.params.token;
+
+    const userData = await model.ResetPassword.findOne({ where: { token: token } });
+    if (!userData) {
+      res.status(404).json({message: "Token has expired"});
+    } else {
+
+      const payload = verifyToken(token, process.env.JWT_SECRET)
+
```

Comment:
How do you know the one who is using that token is the one who requested to reset the password token?

for this scenario, if I request to reset my password with my email, then if I copy the token sent to my email and give it to you, you can change your password with my Token right?

Reply:
No! if I use your token then  I will be changing your password not mine, because when user is resetting password the app save the user email and the corresponding token generated for that user then it send the email containing that token, 
then before changing the user password it first check if token exist in reset password model then it change the password of the user who has that token